src/Features/CSharpTest/ExtractMethod/ExtractMethodTests2.cs

```diff
@@ -5867,6 +5868,101 @@ public async Task Goo()
             """,
             options: ImplicitTypeEverywhere());
 
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/61555")]
+    public async Task TestKnownNotNullParameter()
+        => await new VerifyCS.Test
+        {
+            TestCode = """
+                #nullable enable
+
+                public class C
+                {
+                    public void M(C? c)
+                    {
+                        if (c == null)
+                        {
+                            return;
+                        }
+
+                        [|c.ToString();|]
+                    }
+                }
+                """,
+            FixedCode = """
+                #nullable enable
+            
+                public class C
+                {
+                    public void M(C? c)
+                    {
+                        if (c == null)
+                        {
+                            return;
+                        }
+            
+                        NewMethod(c);
+                    }
+
+                    private static void NewMethod(C c)
+                    {
+                        c.ToString();
+                    }
+                }
+                """,
+            LanguageVersion = LanguageVersion.CSharp13,
+            ReferenceAssemblies = ReferenceAssemblies.Net.Net90,
+        }.RunAsync();
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/61555")]
+    public async Task TestKnownNotNullParameter_AssignedNull()
+        => await new VerifyCS.Test
+        {
+            TestCode = """
+                #nullable enable
+
+                public class C
+                {
+                    public void M(C? c)
+                    {
+                        if (c == null)
+                        {
+                            return;
+                        }
+
+                        [|c.ToString();
+                        c = null;
+                        c?.ToString();|]
+                    }
+                }
+                """,
+            FixedCode = """
+                #nullable enable
+            
+                public class C
+                {
+                    public void M(C? c)
+                    {
+                        if (c == null)
+                        {
+                            return;
+                        }
+            
+                        c = NewMethod(c);
+                    }
+
+                    private static C? NewMethod(C? c)
+                    {
+                        {|CS8602:c|}.ToString();
+                        c = null;
+                        c?.ToString();
+                        return c;
```

Comment:
Why return `c` if the extracted selection contains the last reference to it?

Reply:
Unclear :-). I'll have to debug through to see what's going on (note, this is unrelated to this pr).